### DISCUSSION TITLE
Extension Main Class Cannot Be Found

### DISCUSSION THREAD

**Discussion by amnoah** [Minestom/Minestom#1331] [Help]
I've been trying to port a plugin of mine to Minestom, but I can't seem to get it to start properly.
Upon starting I receive the error `[20:23:42 ERROR] [main] Could not find main class 'am.noah.benchmark.minestom.BenchmarkPlugin' in extension 'benchmark'.`.

My extension.json:
```
{
  "entrypoint": "am.noah.benchmark.minestom.BenchmarkPlugin",
  "name": "benchmark",
  "version": "0.1"
}
```

Below is a screenshot of the module, and as far as I can tell my extension.json matches the module layout.
![image](https://user-images.githubusercontent.com/79341053/184726403-18e7d451-9b37-47e0-9d26-bd71b2b5c558.png)

And yes, I have searched for similar discussion posts, but unfortunately the only similar one hasn't been answered.

**Comment by MrlnHi** [comment]
Open your built jar in a zip program (e.g. 7zip) and look at how the structure is there. There might be something wrong with your Maven configuration

**Reply by amnoah** [reply]
Unfortunately I see no difference, it seems identical?
![image](https://user-images.githubusercontent.com/79341053/185024036-5e1b7d6b-d1c9-47fa-bb8e-bc78bba5dbd2.png)

**Comment by dev-hydrogen** [comment]
Does the class extend the Extension class? (If it does, check that its the correct extension class and not some other class named Extension)

**Comment by MrlnHi** [comment]
Please share your BenchmarkPlugin class (if you can)
